src/radical/pilot/agent/lm/prte.py

```diff
@@ -0,0 +1,284 @@
+        task_exec    = cud['executable']
+        task_env     = cud.get('environment') or dict()
+        task_args    = cud.get('arguments')   or list()
+        task_argstr  = self._create_arg_string(task_args)
+
+        n_threads = cu['description'].get('cpu_threads',   1)
+        n_procs   = cu['description'].get('cpu_processes', 0) \
+                  + cu['description'].get('gpu_processes', 0)
+
+        if not n_procs  : n_procs   = 1
+        if not n_threads: n_threads = 1
+
+      # import pprint
+      # self._log.debug('=== prep %s', pprint.pformat(cu))
+        self._log.debug('prep %s', cu['uid'])
+
+        if 'lm_info' not in slots:
+            raise RuntimeError('No lm_info to launch via %s: %s'
+                               % (self.name, slots))
+
+        if not slots['lm_info']:
+            raise RuntimeError('lm_info missing for %s: %s'
+                               % (self.name, slots))
+
+        if 'dvm_uri' not in slots['lm_info']:
+            raise RuntimeError('dvm_uri not in lm_info for %s: %s'
+                               % (self.name, slots))
+
+        dvm_uri = slots['lm_info']['dvm_uri']
+
+        if task_argstr: task_command = "%s %s" % (task_exec, task_argstr)
+        else          : task_command = task_exec
+
+        env_string = ''
+        env_list   = self.EXPORT_ENV_VARIABLES + task_env.keys()
+        if env_list:
+            for var in env_list:
+                env_string += '-x "%s" ' % var
+
+        if 'nodes' not in slots:
+            # this task is unscheduled - we leave it to PRRTE/PMI-X to
+            # correctly place the task.  Just count procs and threads.
+
+            map_flag  = ' --bind-to none -use-hwthread-cpus --oversubscribe'
+            map_flag += ' -n %d --cpus-per-proc %d' % (n_procs, n_threads)
+
+        else:
+            # enact the scheduler's placement
+            hosts_string = ''
+            depths       = set()
+            for node in slots['nodes']:
+                self._log.debug('=== %s' % node)
+
+                # add all cpu and gpu process slots to the node list.
+                for cpu_slot in node[2]: hosts_string += '%s,' % node[0]
+                for gpu_slot in node[3]: hosts_string += '%s,' % node[0]
+                for cpu_slot in node[2]: depths.add(len(cpu_slot))
+
+            # remove trailing ','
+            hosts_string = hosts_string.rstrip(',')
+
+            # FIXME: ensure correct binding for procs and threads
+          # assert(len(depths) == 1), depths
+          # depth = depths.pop()
+          # if depth > 1: map_flag = '--bind-to none --map-by ppr:%d:core' % depth
+          # else        : map_flag = '--bind-to none'
+            map_flag  = ' --bind-to none -use-hwthread-cpus --oversubscribe'
+            map_flag += ' -n %d --cpus-per-proc %d' % (n_procs, n_threads)
+            map_flag += ' -host %s' % hosts_string
+
+
+        # Additional (debug) arguments to prun
+        if True or self._verbose:
```

Comment:
This is always true, right? If this is the intended behavior, I think removing the `if` makes sense, otherwise removing `True` makes sense

Reply:
Ah, thanks - I removed the `True` ...  Thanks!